core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -838,7 +849,7 @@ protected Jenkins(File root, ServletContext context, PluginManager pluginManager
         oldJenkinsJVM = JenkinsJVM.isJenkinsJVM(); // capture to restore in cleanUp()
         JenkinsJVMAccess._setJenkinsJVM(true); // set it for unit tests as they will not have gone through WebAppMain
         long start = System.currentTimeMillis();
-
+        STARTUP_MARKER_FILE = new FileBoolean(new File(root, ".lastStarted"));
```

Comment:
Is this file ever removed? In a classic Jenkins instance it will always stay there after the first start. Doesn't it involves too much (adding an extra file for ever) for just a level in the warning? Isn't there any other way to catch we are in a fresh startup? I.e: the former build or workspace directories already exists or something like that.

Reply:
I believe 1) adding marker file is a simple enough solution and commonly used in Jenkins codebase (see the `secret.key.not-so-secret` file for instance and 2) this _could_ be useful for other code. Low-tech is sometimes also good too IMO: if something goes wrong, we can ask user to delete (or create) this file to trigger/revert to some other behavior.

Reply:
I agree low-tech is good, almost always. I was just asking if there were any other approach, with the existing infra (files, directories, status files)